bokeh/layouts.py

```diff
@@ -75,7 +75,12 @@
 # General API
 #-----------------------------------------------------------------------------
 
-def row(*args: LayoutDOM, children: List[LayoutDOM] | None = None, sizing_mode: SizingModeType | None = None, **kwargs) -> Row:
+@overload
+def row(children: List[LayoutDOM], *, sizing_mode: SizingModeType | None = None, **kwargs: Any) -> Row: ...
+@overload
+def row(*children: LayoutDOM, sizing_mode: SizingModeType | None = None, **kwargs: Any) -> Row: ...
```

Comment:
Overloading is useful when return types vary according to input:

https://mypy.readthedocs.io/en/stable/more_types.html#function-overloading

What is the purpose of using overloads here where there is no such relationship?

Reply:
That's one class of signatures where overloads help. Another, as used here, helps us to disallow mutually exclusive combinations of arguments. In this case we want to disallow `row(Plot(), children=[Plot()])`. Previously this could have been verified only at run time (it still is verified).

Reply:
> it still is verified

You have removed runtime checks in other instances, what is your decision criteria for leaving in vs removing?

Reply:
I always leave them, because the goal of this work is not to change run time logic and at run time, overloads collapse into a single signature to, so you have to analyze this anyway. But at least, when using this function is used in a typed context, one will know what combinations make sense and which don't.

Reply:
looking back, the cases where you removed runtime checks was for cases where function signatures that were changed